### PR TITLE
wkd: Initial commit

### PR DESCRIPTION
**Pull request by drGrove**
Add support for wkd running at openpgpkey.hashbang.sh

### CODE REVIEW

**Comment by RyanSquared** [comment]
why have each key be a separate configmap?

**Comment by drGrove** [comment]
> why have each key be a separate configmap?

Configmaps/labels have a max size. We'll be able to decrease duplication in the future when both mtls and wkd can read from the same place

**Comment by RyanSquared** [comment]
> Configmaps/labels have a max size.

TIL; I thought it was per value.

**Comment by drGrove** [comment]
> > Configmaps/labels have a max size.
> 
> TIL; I thought it was per value.

It's less the configmap and more that annotations are used to show previous content for comparisons and it's not great

File: wkd/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: wkd
+resources:
+  - github.com/drGrove/docker-wkd/kustomize/direct?ref=7d81b14e809401b8b163b3278e41254b1a05b2be
+  - ingress.yaml
+patchesJson6902:
```
Comment (daurnimator) [inline wkd/kustomization.yaml:7]: just use `patches:`

File: wkd/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  - target:
+      group: apps
+      version: v1
```
Comment (daurnimator) [inline wkd/kustomization.yaml:10]: who *really* needs `group` or `version`? :)
```suggestion
```

File: wkd/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      kind: Deployment
+      name: wkd
+    patch: |-
```
Comment (daurnimator) [inline wkd/kustomization.yaml:13]: why use a 6092 patch instead of a strategic merge patch?

File: wkd/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      - op: add
+        path: "/spec/template/spec/containers/0/args/-"
+        value: "-m"
+      - op: add
+        path: "/spec/template/spec/containers/0/args/-"
+        value: "hashbang.sh"
+images:
+  - name: drgrove/wkd
+    digest: sha256:b00e9e42486efa6e8601f4aff3d61c9417dcaba993411905a4b63cd2dee72f42
```
Comment (daurnimator) [inline wkd/kustomization.yaml:22]: add version number as a comment or something next to this

File: wkd/kustomization.yaml (hunk 3 above)
Comment (drGrove) [reply wkd/kustomization.yaml:13]: Patch Strategic Merge does not work with adding to an array, unless you override the entire array

**Review by daurnimator — CHANGES_REQUESTED** [review]

File: argocd/applications/wkd.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Application
```
Comment (daurnimator) [inline argocd/applications/wkd.yaml:2]: you don't need an application any more.

File: wkd/ingress.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: openpgpkey-hashbang-sh
+  labels:
+    app.kubernetes.io/name: openpgpkey-hashbang-sh
+  annotations:
+    cert-manager.io/cluster-issuer: letsencrypt-prod
+    kubernetes.io/ingress.class: nginx
```
Comment (daurnimator) [inline wkd/ingress.yaml:9]: That's the old annotation

File: wkd/kustomization.yaml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      - op: add
+        path: "/spec/template/spec/containers/0/args/-"
+        value: "-m"
+      - op: add
+        path: "/spec/template/spec/containers/0/args/-"
+        value: "hashbang.sh"
+images:
+  - name: drgrove/wkd
+    digest: sha256:b00e9e42486efa6e8601f4aff3d61c9417dcaba993411905a4b63cd2dee72f42 # v2.2.2
+configMapGenerator:
```
Comment (daurnimator) [inline wkd/kustomization.yaml:21]: where's my key? :(

**Review by daurnimator — CHANGES_REQUESTED** [review]

File: wkd/kustomization.yaml (hunk 7 above)
Comment (drGrove) [reply wkd/kustomization.yaml:21]: You don't have an @hasbang.sh ID

File: wkd/kustomization.yaml [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+      kind: Deployment
+      name: wkd
+    patch: |-
+      - op: add
+        path: "/spec/template/spec/containers/0/args/-"
+        value: "-m"
+      - op: add
+        path: "/spec/template/spec/containers/0/args/-"
+        value: "hashbang.sh"
+images:
```
Comment (daurnimator) [inline wkd/kustomization.yaml:18]: Also need to hashlock the sidecar

**Review by daurnimator — CHANGES_REQUESTED** [review]

File: wkd/kustomization.yaml [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+  - name: drgrove/wkd
+    digest: sha256:b00e9e42486efa6e8601f4aff3d61c9417dcaba993411905a4b63cd2dee72f42 # v2.2.2
+  - name: kiwigrid/k8s-sidecar
+    digest: sha256:9b6e173c61ca4a9470a52729f362dabd5962581d85d9e6746dcfb9fcfa484b42 # 1.21.0
```
Comment (daurnimator) [inline wkd/kustomization.yaml:22]:
```suggestion
  - name: kiwigrid/k8s-sidecar:1.21.0
    digest: sha256:9b6e173c61ca4a9470a52729f362dabd5962581d85d9e6746dcfb9fcfa484b42
```

by doing it this way we can make sure we don't accidentally downgrade on the next bump

**Review by daurnimator — APPROVED** [review]

File: wkd/kustomization.yaml (hunk 9 above)
Comment (drGrove) [reply wkd/kustomization.yaml:22]: I don't think this will work because the upstream has `:latest`

**Pull request merged**